spec.html

```diff
@@ -8034,11 +8034,10 @@ <h1>ArraySetLength ( _A_, _Desc_ )</h1>
             1. Set _newLenDesc_.[[Writable]] to *true*.
           1. Let _succeeded_ be ! OrdinaryDefineOwnProperty(_A_, `"length"`, _newLenDesc_).
           1. If _succeeded_ is *false*, return *false*.
-          1. Repeat, while _newLen_ &lt; _oldLen_,
-            1. Set _oldLen_ to _oldLen_ - 1.
-            1. Let _deleteSucceeded_ be ! _A_.[[Delete]](! ToString(_oldLen_)).
+          1. For each own property key _P_ of _A_ that is an array index and is greater than or equal to _newLen_, in descending numeric index order, do
```

Comment:
prose seems like a strange way to express this; why not something like (pseudocode):
```
1. Let _ownArrayIndices_ = GetOwnArrayIndices(_A_).
1. For each _P_ in _ownArrayIndices_ in List order, do
```
?

Reply:
I based it on `For each own property key _P_ of _A_ that is an array index, in ascending numeric index order` from OrdinaryOwnPropertyKeys